konrad/atmosphere.py

```diff
@@ -377,6 +377,34 @@ def get_cold_point_plev(self, interpolate=False):
             # Return the single coldest point on the actual pressure grid.
             return self["plev"][self.get_cold_point_index()]
 
+    def get_tropopause_index_wmo(self):
+        self.calculate_height()
+        height = self["z"][-1, :] / 1000
+        lapse_rate = self.get_lapse_rates() * (-1000)
+        height_diffs = np.diff(height)
+        height_diffs = np.insert(height_diffs, 0, 0)
+        for i in np.ma.masked_array(height, (lapse_rate > 2)).nonzero()[0]:
+            current_height = height[i]
+            test_height = np.ma.masked_outside(
+                height, current_height, current_height + 2.5
+            )
+
+            if (
+                np.ma.average(
+                    np.ma.masked_array(lapse_rate, test_height.mask),
+                    weights=height_diffs,
+                )
+            ) < 2 and (i > 0):
+                return i
+
+        return None
```

Comment:
Maybe one should reconsider the choice of the return value for cases where no tropopause is found. The current value (`None`) will result in returning the full array, which is likely to break other code that expects a float only.

I would either opt for a valid value, e.g., model top (`-1`) or raising a proper error.

Reply:
would it be okay to do the second, but give a warning that no proper value has been found? 
Or should there be an option to take the cold-point if nothing is found?

Reply:
The error message should of course give a reasonable hint of what happened, e.g.:

```py
raise RuntimeError("Could not find a tropopause based on WMO definition.")
```

I wouldn't fallback to the cold-point in these situations. Actually, I don't think that this will happen at all, because the WMO definition is quite robust. This is more a nitpicky technical point that **if** something unexpected happens, konrad should behave in a sane way :)